### PR TITLE
Docs - Guide to add a new TensorFlow model

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Adds a guide on how to add a new TensorFlow model (architecture and weights), as well as a few tips to debug TF<>PT mismatches. This means that we can stop relying on ad hoc tips and instructions to add TF models, and point towards this guide instead 🙌 

I've also updated the test paths in the guide to add new models, which were outdated, and added a cheeky "from" in the "Converting from TensorFlow checkpoints" guide title to clearly distinguish it from the guide I'm adding in this PR.

Review suggestion: read the [live docs](https://moon-ci-docs.huggingface.co/docs/transformers/pr_19256/en/add_tensorflow_model) themselves :)

### CODE REVIEW

File: docs/source/en/add_tensorflow_model.mdx [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+side. If the specific model you want to use with TensorFlow already has a TensorFlow architecture implementation in
+🤗 Transformers but is lacking weights, feel free to jump straight into the weight conversion section of this page.
+
+For simplicity, the remainder of this guide assumes you've decided to contribute with the TensorFlow version of
+*BrandNewBert* (the same example as in the [guide](add_new_model) to add a new model from scratch).
+
+<Tip>
+
+Before starting the work on a TensorFlow model architecture, double-check that there is no ongoing effort to do so.
+You can search for `TensorFlow BrandNewBert` on the [pull request GitHub page](https://github.com/huggingface/transformers/pulls).
```
Comment (sgugger) [inline docs/source/en/add_tensorflow_model.mdx:89]: Might also be just `TF BrandNewBert` sometime people are lazy ;-)

**Review by sgugger — APPROVED** [review]
Very nice, thanks a lot for adding this!

File: docs/source/en/add_tensorflow_model.mdx (hunk 1 above)
Comment (gante) [reply docs/source/en/add_tensorflow_model.mdx]: added a more general tip (search for `BrandNewBert` is enough to find a potential TF PR)

File: docs/source/en/add_tensorflow_model.mdx [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+It's now time to convert your draft pull request into a real pull request. To do so, click on the "Ready for
+review" button and add Joao (`@gante`) and Matt (`@Rocketknight1`) as reviewers. A model pull request will need
+at least 3 reviewers, but they will take care of finding appropriate additional reviewers for your model.
+
+After all reviewers are happy with the state of your PR, the final action point is to remove the `from_pt=True` flag in
+`.from_pretrained()` calls. Since there are no TensorFlow weights, you will have to add them! Check the section
+below for instructions on how to do it.
+
+Finally, when the TensorFlow weights get merged, you have at least 3 reviewer approvals, and all CI checks are
+green, double-check the tests localy one last time
```
Comment (amyeroberts) [inline docs/source/en/add_tensorflow_model.mdx:278]:
```suggestion
green, double-check the tests locally one last time
```

File: docs/source/en/add_tensorflow_model.mdx [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+   containing the PyTorch weights you want to convert
+3. Tag `@joaogante` and `@Rocketknight1` in the 🤗 Hub PR the command above has just created
+
+That's it! 🎉
+
+
+## Debugging mismatches across ML frameworks 🐛
+
+At some point, when adding a new architecture or when creating TensorFlow weights for an existing architecture, you
+might come across errors compaining about mismatches between PyTorch and TensorFlow. You might even decide open the
```
Comment (amyeroberts) [inline docs/source/en/add_tensorflow_model.mdx:318]:
```suggestion
might come across errors complaining about mismatches between PyTorch and TensorFlow. You might even decide open the
```

Comment (amyeroberts) [inline docs/source/en/add_tensorflow_model.mdx:318]:
```suggestion
might come across errors compaining about mismatches between PyTorch and TensorFlow. You might even decide to open the
```

**Review by amyeroberts — APPROVED** [review]
🤩  Amazing - thanks for writing this up! ❤️ 

Comment are just a few spelling and grammar nits

File: docs/source/en/add_tensorflow_model.mdx [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+
+### 1.-3. Prepare your model contribution
+
+**1. Select the model you wish to convert**
+
+Let's start off with the basics: the first thing you need to know is the architecture you want to convert. If you
+haven't your eyes set on a specific architecture, asking the 🤗 Transformers team for suggestions is a great way to
+maximize your impact - we will guide you towards the most prominent architectures that are missing on the TensorFlow
+side. If the specific model you want to use with TensorFlow already has a TensorFlow architecture implementation in
+🤗 Transformers but is lacking weights, feel free to jump straight into the weight conversion section of this page.
```
[the remark this answers is not in the document]
Comment (gante) [reply docs/source/en/add_tensorflow_model.mdx]: added 👍

File: docs/source/en/add_tensorflow_model.mdx [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+
+
+## Adding TensorFlow weights to 🤗 Hub
+
+Assuming that the TensorFlow model architecture is available in 🤗 Transformers, converting PyTorch weights into
+TensorFlow weights is a breeze!
+
+Here's how to do it:
+1. Make sure you are logged into your Hugging Face account in your terminal. You can log in using the command
+   `huggingface-cli login` (you find your access tokens [here](https://huggingface.co/settings/tokens))
```
Comment (gante) [inline docs/source/en/add_tensorflow_model.mdx:309]:
```suggestion
   `huggingface-cli login` (you can find your access tokens [here](https://huggingface.co/settings/tokens))
```

File: docs/source/en/add_tensorflow_model.mdx [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+Whether you wish to have a deeper understanding of large TensorFlow models, make a major open-source contribution, or
+enable TensorFlow for your model of choice, this guide is for you.
+
+This guide empowers you, a member of our community, to contribute with TensorFlow model weights and/or
+architectures to be used in 🤗 Transformers, with minimal supervision from the Hugging Face team. Writing a new model
+is no small feat, but hopefully this guide will make it less of a rollercoaster 🎢 and more of a walk in the park 🚶.
+Harnessing our collective experiences is absolutely critical to make this process increasingly easier, and thus we
+highly encourage that you suggest improvements to this guide!
+
+Before you dive deeper, it is recommended that you check the following resources if you're new in 🤗 Transformers:
```
Comment (Rocketknight1) [inline docs/source/en/add_tensorflow_model.mdx:26]:
```suggestion
Before you dive deeper, it is recommended that you check the following resources if you're new to 🤗 Transformers:
```

File: docs/source/en/add_tensorflow_model.mdx [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+
+# How to convert a 🤗 Transformers model to TensorFlow?
+
+Having multiple frameworks available to use with 🤗 Transformers gives you flexibility to play their strengths when
+designing your application, but it implies that compatibility must be added on a per-model basis. The good news is that
+adding TensorFlow compatibility to an existing model is simpler than [adding a new model from scratch](add_new_model)!
+Whether you wish to have a deeper understanding of large TensorFlow models, make a major open-source contribution, or
+enable TensorFlow for your model of choice, this guide is for you.
+
+This guide empowers you, a member of our community, to contribute with TensorFlow model weights and/or
```
Comment (Rocketknight1) [inline docs/source/en/add_tensorflow_model.mdx:20]:
```suggestion
This guide empowers you, a member of our community, to contribute TensorFlow model weights and/or
```

File: docs/source/en/add_tensorflow_model.mdx [hunk 8]
```diff
@@ -0,0 +1,346 @@
...
+architectures to be used in 🤗 Transformers, with minimal supervision from the Hugging Face team. Writing a new model
+is no small feat, but hopefully this guide will make it less of a rollercoaster 🎢 and more of a walk in the park 🚶.
+Harnessing our collective experiences is absolutely critical to make this process increasingly easier, and thus we
+highly encourage that you suggest improvements to this guide!
+
+Before you dive deeper, it is recommended that you check the following resources if you're new in 🤗 Transformers:
+- [General overview of 🤗 Transformers](add_new_model#general-overview-of-transformers)
+- [Hugging Face's TensorFlow Philosophy](https://huggingface.co/blog/tensorflow-philosophy)
+
+For the remainder of this guide, you will learn what's needed to add a new TensorFlow model architecture, the
```
Comment (Rocketknight1) [inline docs/source/en/add_tensorflow_model.mdx:30]:
```suggestion
In the remainder of this guide, you will learn what's needed to add a new TensorFlow model architecture, the
```

File: docs/source/en/add_tensorflow_model.mdx [hunk 9]
```diff
@@ -0,0 +1,346 @@
...
+
+After you're done, it's time for the moment of truth: run the tests! 😬
+
+```bash
+NVIDIA_TF32_OVERRIDE=0 RUN_SLOW=1 RUN_PT_TF_CROSS_TESTS=1 \
+py.test -vv tests/models/brand_new_bert/test_modeling_tf_brand_new_bert.py
+```
+
+The most likely outcome is that you'll see a bunch of errors. Don't worry, this is expected! Debugging ML models is
+notoriously hard, and the key ingredient to success is patience (and `breakpoint()`). In our experience, the hardest
```
Comment (Rocketknight1) [inline docs/source/en/add_tensorflow_model.mdx:255]: :pray:

**Review by Rocketknight1 — APPROVED** [review]
This is a really excellent guide! I have a couple of grammar nitpicks, but they're not even errors, they're just style choices more than anything.

**Comment by gante** [comment]
Awesome! I will merge the PR soon and make some guide-related announcements next Monday 🙌

**Pull request merged**